preprocessors/togeojson-gpx.preprocessor.js

```diff
@@ -0,0 +1,156 @@
+var gdal = require('gdal');
+var fs = require('fs');
+var mkdirp = require('mkdirp');
+var queue = require('queue-async');
+var spawn = require('child_process').spawn;
+var path = require('path');
+var digest = require('mapnik-omnivore').digest;
+var mapnik = require('mapnik');
+var mapnik_index = path.resolve(mapnik.module_path, 'mapnik-index' + (process.platform === 'win32' ? '.exe' : ''));
+if (!fs.existsSync(mapnik_index)) {
+  throw new Error('mapnik-index does not exist at ' + mapnik_index);
+}
+
+//disable in production
+//gdal.verbose();
+
+module.exports = function(infile, outdirectory, callback) {
+  mkdirp(outdirectory, function(err) {
+    if (err) return callback(err);
+
+    var geojson_files = [];
+    var ds_gpx;
+    var full_feature_cnt = 0;
+    var wgs84 = gdal.SpatialReference.fromEPSG(4326);
+
+    try {
+      ds_gpx = gdal.open(infile);
+    }
+    catch (err) {
+      return callback(new Error(err));
+    }
+
+    ds_gpx.layers.forEach(function(lyr_gpx) {
+      //drop point layers as they can get really huge
+      if (lyr_gpx.name === 'track_points' || lyr_gpx.name === 'route_points') {
```

Comment:
Does this represent a breaking change from present behavior? That is, would these layers be included in a GPX upload today?